clang/unittests/Analysis/FlowSensitive/NullCheckAfterDereferenceModelTest.cpp

```diff
@@ -0,0 +1,343 @@
+//===- NullCheckAfterDereferenceModelTest.cpp -------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+//  This file defines a test for pointer nullability, specifically focused on
+//  finding invalid dereferences, and unnecessary null-checks.
+//  Only a limited set of operations are currently recognized. Notably, pointer
+//  arithmetic, null-pointer assignments and _nullable/_nonnull attributes are
+//  missing as of yet.
+//
+//  FIXME: Port over to the new type of dataflow test infrastructure
```

Comment:
Still pending, and I would like to write a couple more simple tests in the new testing infrastructure as well.
(The new infra uses tests where instead of checking for a specific null/nonnull state afterwards, the test marks the expected null-state within the inline comments, leading to more readable tests.)